### PR TITLE
Lytt linux tests

### PR DESCRIPTION
**Pull request by Kadino**
Python unit tests passing on Linux, including a basic implementation of the Linux platform-manager. Includes a few windows-specific fixes to compensate for initial Linux-fixes.

This does not yet enable end-to-end editor testing functionality on Linux, but incrementally steps forward into working unit tests. I plan to immediately work on enabling things end-to-end.

Wanted to remove the winreg usage altogether, but fell back on simply updating its import pattern. It should not be removed yet, as there are still hooks in a few tools accessing the OS-specific registry instead of the new settings-registry pattern.

### CODE REVIEW

File: Tools/LyTestTools/tests/unit/test_file_system.py [hunk 1]
```diff
@@ -208,20 +208,6 @@ def test_Unzip_ReleaseBuild_ReturnsCorrectPath(self, mock_join, mock_check_free)
 
         self.assertEqual(path, expected_path)
 
-    @mock.patch('os.path.exists')
```
Comment (Kadino) [inline Tools/LyTestTools/tests/unit/test_file_system.py:211]: These zip/unzip utilities have not been used by tests in ages. Paths resolve differently between Windows and non-windows, so the mock-assert is invalid outside of Windows. It was easier to remove these low-value tests.

File: Tools/LyTestTools/ly_test_tools/launchers/platforms/linux/launcher.py [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+            'Project cannot be NoneType - please specify a project name string.')
+        return os.path.join(f"{self.workspace.paths.build_directory()}",
+                            f"{self.workspace.project}.ServerLauncher")
+
+
+class LinuxEditor(LinuxLauncher):
+
+    def __init__(self, build, args):
+        super(LinuxEditor, self).__init__(build, args)
+        self.args.append('--regset="/Amazon/Settings/EnableSourceControl=false"')
```
Comment (Kadino) [inline Tools/LyTestTools/ly_test_tools/launchers/platforms/linux/launcher.py:213]: I expect these args are invalid for Linux, will verify during the next phase of investigation

File: Tools/LyTestTools/ly_test_tools/_internal/managers/platforms/linux.py [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+    def editor_log(self):
+        """
+        :return: path to editor.log
+        """
+        return os.path.join(self.project_log(), "editor.log")
+
+
+class LinuxWorkspaceManager(AbstractWorkspaceManager):
+    """
+    A Mac host WorkspaceManager. Contains Mac overridden functions for the AbstractWorkspaceManager class.
```
Comment (AMZN-scspaldi) [inline Tools/LyTestTools/ly_test_tools/_internal/managers/platforms/linux.py:62]: Copy/paste error in the docstring?

File: Tools/LyTestTools/ly_test_tools/environment/process_utils.py [hunk 4]
```diff
@@ -12,8 +12,8 @@
 import subprocess
 import ctypes
 
+import ly_test_tools
 import ly_test_tools.environment.waiter as waiter
-from ly_test_tools import WINDOWS, MAC
```
Comment (AMZN-scspaldi) [inline Tools/LyTestTools/ly_test_tools/environment/process_utils.py:16]: Shouldn't this be from ly_test_tools import LINUX, MAC, WINDOWS to match the file above? Why two different patterns?

File: Tools/LyTestTools/ly_test_tools/environment/watchdog.py [hunk 5]
```diff
@@ -138,8 +138,8 @@ def __init__(self, process_id, interval=1, raise_on_condition=True, name='proces
         :param unresponsive_timeout_seconds: How long the process needs to be unresponsive for in order for the watchdog
         to report (in seconds).
         """
-        if not WINDOWS:
-            raise (NotImplementedError, "Process watchdog is only implemented on Windows.")
+        if not ly_test_tools.WINDOWS:
+            pass # TODO add non-windows support
```
Comment (AMZN-scspaldi) [inline Tools/LyTestTools/ly_test_tools/environment/watchdog.py:142]: Are we still supposed to link to issues for to-dos?

File: Tools/LyTestTools/ly_test_tools/launchers/platforms/linux/launcher.py [hunk 6]
```diff
@@ -0,0 +1,224 @@
...
+        """
+        if self._proc:
+            return self._proc.poll()
+        else:
+            return self._ret_code
+
+    def check_returncode(self):
+        # type: () -> None
+        """
+        Checks the returncode of the launcher if it exists. Raises a CrashError if the returncode is non-zero. Returns
```
Comment (AMZN-scspaldi) [inline Tools/LyTestTools/ly_test_tools/launchers/platforms/linux/launcher.py:150]: nit: return code

**Review by AMZN-scspaldi — CHANGES_REQUESTED** [review]
Are you testing these locally or on an instance?

File: Tools/LyTestTools/ly_test_tools/__init__.py [hunk 7]
```diff
@@ -51,9 +51,12 @@
     from ly_test_tools.launchers import MacLauncher
     LAUNCHERS['mac'] = MacLauncher
 elif LINUX:
-    logger.warning(f'Linux operating system is currently not supported, LyTestTools only supports Windows and Mac.')
     HOST_OS_PLATFORM = 'linux'
-    HOST_OS_EDITOR = NotImplementedError('LyTestTools does not yet support Linux editor')
-    HOST_OS_DEDICATED_SERVER = NotImplementedError('LyTestTools does not yet support Linux dedicated server')
+    HOST_OS_EDITOR = 'linux_editor'
```
[the remark this answers is not in the document]
Comment (Kadino) [reply Tools/LyTestTools/ly_test_tools/__init__.py:55]: Likely yes, but that will be part of the next step and/or a feature request.

File: Tools/LyTestTools/ly_test_tools/environment/process_utils.py (hunk 4 above)
Comment (Kadino) [reply Tools/LyTestTools/ly_test_tools/environment/process_utils.py:16]: The pattern should be made consistent across tools, to import ly_test_tools and inspect its members, not copy their values. I'm trying to make the pattern consistent but would not be surprised that there are still extraneous from-imports

File: Tools/LyTestTools/ly_test_tools/environment/watchdog.py (hunk 5 above)
Comment (Kadino) [reply Tools/LyTestTools/ly_test_tools/environment/watchdog.py:142]: I do not plan on linking an internally-tracked issue, and we were blocked from moving these tasks into GHI. The issue is the immediate follow up task.

**Comment by Kadino** [comment]
> 
> 
> Are you testing these locally or on an instance?

These changes were tested both on a remote EC2 Ubuntu instance, as well as locally on a Windows instance.

**Review by AMZN-scspaldi — APPROVED** [review]

**Review by evanchia-ly-sdets — APPROVED** [review]

**Pull request merged**